packages/stack/test-runner/src/lib/index.js

```diff
@@ -52,14 +53,13 @@ class TestRunner {
     this.events.setCommandHandler('tests:deployment:check', this.checkDeploymentOptions.bind(this));
     this.events.setCommandHandler('tests:blockchain:start', this.startBlockchainNode.bind(this));
     this.events.setCommandHandler('tests:config:check', this.checkModuleConfigs.bind(this));
+    this.events.setCommandHandler('tests:globals:setup', this.setupGlobalVariables.bind(this));
   }
 
   run(options, cb) {
     const reporter = new Reporter(this.embark, {stdout: this.stdout});
     const testPath = options.file || "test";
 
-    this.setupGlobalVariables();
```

Comment:
Alright, this is something I'd like to get some feedback on:

`setupGlobalVariables()` used to be sync, but is now async because it needs to request information from the connected node (so we can expose it in the tests).

I had to remove it from here and have the test-runner plugin request the setup [here](https://github.com/embark-framework/embark/pull/2169/files#diff-d7fd409f41e4b03793bca97eb99ad303R159) though. The reason for that is that when we try getting a web3 instance using the test-runner's getter, it will eventually request an endpoint from the proxy.

The proxy however will never respond because it only sets up its `proxy:endpoint:get` endpoint **after** the blockchain has emitted `blockchain:started`. The latter is actually not happening when the simulator is used. Also, the blockchain is actually only started when the test-runner plugin's (e.g. Mocha) `run()` function is called, which  also happens later.

Therefore, I decided to move responsibility of setting up globals to the test-runner plugin by requesting `tests:globals:setup` at a point where we know for sure the blockchain is running and we can successfully request information about it.

Any thoughts ideas on how to do it differently are very welcome.

Reply:
Short of a heavier refactor, this seems like a decent solution, nice work.  Longer term, I think we should think about how to make the setup steps (e.g. starting the blockchain) as homogeneous as possible across `run`, `test`, etc.